earth_enterprise/src/common/performancelogger.cpp

```diff
@@ -57,3 +57,12 @@ void PerformanceLogger::log(
 
   notify(NFY_NOTICE, "%s\n", message.str().c_str());
 }
+
+// Thread safety wrapper for log output
+PerformanceLogger::do_notify( const string& message, ostream& out, khMutex& mutex ) {
+
+  mutex.lock();  // wait for and then lock the mutex
```

Comment:
I believe the GEE way of doing this is:

```
  {
    khLockGuard lock(mutex);

    out << message << endl;
  }
```

This way, the mutex won't remain locked if there's an exception during output (e.g., disk full, etc.), which can cause threads to remain deadlocked.